File: suc/kotlin/076.kt
package com.example.myapplication

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Home
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Menu
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.rounded.Check
import androidx.compose.material.icons.rounded.Home
import androidx.compose.material.icons.rounded.Lock
import androidx.compose.material.icons.rounded.Notifications
import androidx.compose.material.icons.rounded.Star
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        val windowInsetsController = WindowCompat.getInsetsController(window, window.decorView)
        windowInsetsController.systemBarsBehavior =
            WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        windowInsetsController.hide(WindowInsetsCompat.Type.systemBars())

        setContent {
            MaterialTheme(
                colorScheme = androidx.compose.material3.lightColorScheme()
            ) {
                SmartHomeDashboard()
            }
        }
    }
}

@Composable
fun SmartHomeDashboard() {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFFF4F6F8))
            .padding(24.dp)
            .verticalScroll(rememberScrollState())
    ) {
        Spacer(modifier = Modifier.height(16.dp))

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column {
                Text(
                    text = "Good Morning,",
                    fontSize = 16.sp,
                    color = Color.Gray,
                    fontWeight = FontWeight.Medium
                )
                Text(
                    text = "Alex Johnson",
                    fontSize = 24.sp,
                    color = Color.Black,
                    fontWeight = FontWeight.Bold
                )
            }
            Box(
                modifier = Modifier
                    .size(48.dp)
                    .clip(CircleShape)
                    .background(Color.White),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = Icons.Filled.Person,
                    contentDescription = null,
                    tint = Color.Black
                )
            }
        }

        Spacer(modifier = Modifier.height(32.dp))

        Card(
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(24.dp),
            colors = CardDefaults.cardColors(containerColor = Color(0xFF5E60CE)),
            elevation = CardDefaults.cardElevation(defaultElevation = 8.dp)
        ) {
            Row(
                modifier = Modifier
                    .padding(24.dp)
                    .fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column {
                    Text(
                        text = "Energy Usage",
                        color = Color.White.copy(alpha = 0.8f),
                        fontSize = 14.sp
                    )
                    Text(
                        text = "28.5 kWh",
                        color = Color.White,
                        fontSize = 32.sp,
                        fontWeight = FontWeight.Bold
                    )
                    Text(
                        text = "Today",
                        color = Color.White.copy(alpha = 0.6f),
                        fontSize = 12.sp
                    )
                }
                Box(
                    modifier = Modifier
                        .size(56.dp)
                        .clip(CircleShape)
                        .background(Color.White.copy(alpha = 0.2f)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Rounded.Notifications,
                        contentDescription = null,
                        tint = Color.White
                    )
                }
            }
        }

        Spacer(modifier = Modifier.height(32.dp))

        Text(
            text = "Scenes",
            fontSize = 20.sp,
            fontWeight = FontWeight.Bold,
            color = Color.Black
        )

        Spacer(modifier = Modifier.height(16.dp))

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            SceneButton("Home", Icons.Rounded.Home, true)
            SceneButton("Away", Icons.Rounded.Lock, false)
            SceneButton("Sleep", Icons.Rounded.Star, false)
        }

        Spacer(modifier = Modifier.height(32.dp))

        Text(
            text = "Devices",
            fontSize = 20.sp,
            fontWeight = FontWeight.Bold,
            color = Color.Black
        )

        Spacer(modifier = Modifier.height(16.dp))

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Column(modifier = Modifier.weight(1f)) {
                DeviceCard(
                    name = "Smart Lamp",
                    icon = Icons.Rounded.Star,
                    isActive = true,
                    accentColor = Color(0xFFFFC107)
                )
                Spacer(modifier = Modifier.height(16.dp))
                DeviceCard(
                    name = "Thermostat",
                    icon = Icons.Filled.Menu,
                    isActive = true,
                    accentColor = Color(0xFFFF7043)
                )
            }
            Column(modifier = Modifier.weight(1f)) {
                DeviceCard(
                    name = "Main Lock",
                    icon = Icons.Rounded.Lock,
                    isActive = false,
                    accentColor = Color(0xFF4CAF50)
                )
                Spacer(modifier = Modifier.height(16.dp))
                DeviceCard(
                    name = "Wi-Fi Router",
                    icon = Icons.Filled.Settings,
                    isActive = true,
                    accentColor = Color(0xFF2196F3)
                )
            }
        }

        Spacer(modifier = Modifier.height(24.dp))
    }
}

@Composable
fun SceneButton(text: String, icon: ImageVector, isSelected: Boolean) {
    val backgroundColor = if (isSelected) Color(0xFF212121) else Color.White
    val contentColor = if (isSelected) Color.White else Color.Black

    Box(
        modifier = Modifier
            .clip(RoundedCornerShape(30.dp))
            .background(backgroundColor)
            .clickable { }
            .padding(horizontal = 24.dp, vertical = 12.dp)
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = contentColor,
                modifier = Modifier.size(18.dp)
            )
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = text,
                color = contentColor,
                fontWeight = FontWeight.Medium,
                fontSize = 14.sp
            )
        }
    }
}

@Composable
fun DeviceCard(name: String, icon: ImageVector, isActive: Boolean, accentColor: Color) {
    var switchState by remember { mutableStateOf(isActive) }

    Card(
        shape = RoundedCornerShape(20.dp),
        colors = CardDefaults.cardColors(containerColor = Color.White),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
        modifier = Modifier
            .fillMaxWidth()
            .aspectRatio(0.85f)
    ) {
        Column(
            modifier = Modifier
                .padding(16.dp)
                .fillMaxSize(),
            verticalArrangement = Arrangement.SpaceBetween
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Box(
                    modifier = Modifier
                        .size(40.dp)
                        .clip(CircleShape)
                        .background(if (switchState) accentColor.copy(alpha = 0.1f) else Color(0xFFF0F0F0)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = icon,
                        contentDescription = null,
                        tint = if (switchState) accentColor else Color.Gray,
                        modifier = Modifier.size(24.dp)
                    )
                }
            }

            Column {
                Text(
                    text = name,
                    fontWeight = FontWeight.Bold,
                    fontSize = 16.sp,
                    color = Color.Black
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    text = if (switchState) "On" else "Off",
                    color = if (switchState) accentColor else Color.Gray,
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Medium
                )
            }

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.End
            ) {
                Switch(
                    checked = switchState,
                    onCheckedChange = { switchState = it },
                    colors = SwitchDefaults.colors(
                        checkedThumbColor = Color.White,
                        checkedTrackColor = accentColor,
                        uncheckedThumbColor = Color.White,
                        uncheckedTrackColor = Color(0xFFE0E0E0)
                    ),
                    modifier = Modifier.scale(0.8f)
                )
            }
        }
    }
}

fun Modifier.scale(scale: Float): Modifier {
    return this.then(
        Modifier.size(width = 52.dp * scale, height = 32.dp * scale)
    )
}

File: suc/swift/076.swift
import SwiftUI

@main
struct SmartHomeApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

struct ContentView: View {
    var body: some View {
        SmartHomeDashboard()
            .statusBarHidden(true)
    }
}

struct SmartHomeDashboard: View {
    var body: some View {
        ZStack {
            Color(red: 0.957, green: 0.965, blue: 0.973) // #F4F6F8
                .ignoresSafeArea()
            
            ScrollView(showsIndicators: false) {
                VStack(alignment: .leading, spacing: 0) {
                    
                    // Top Header Spacer
                    Spacer().frame(height: 16)
                    
                    // Header
                    HStack(alignment: .center) {
                        VStack(alignment: .leading, spacing: 0) {
                            Text("Good Morning,")
                                .font(.system(size: 16, weight: .medium))
                                .foregroundColor(.gray)
                            Text("Alex Johnson")
                                .font(.system(size: 24, weight: .bold))
                                .foregroundColor(.black)
                        }
                        
                        Spacer()
                        
                        ZStack {
                            Circle()
                                .fill(Color.white)
                                .frame(width: 48, height: 48)
                            
                            Image(systemName: "person.fill")
                                .resizable()
                                .aspectRatio(contentMode: .fit)
                                .frame(width: 24, height: 24)
                                .foregroundColor(.black)
                        }
                    }
                    .padding(.horizontal, 24)
                    
                    Spacer().frame(height: 32)
                    
                    // Energy Usage Card
                    VStack {
                        HStack(alignment: .center) {
                            VStack(alignment: .leading, spacing: 0) {
                                Text("Energy Usage")
                                    .font(.system(size: 14))
                                    .foregroundColor(.white.opacity(0.8))
                                
                                Text("28.5 kWh")
                                    .font(.system(size: 32, weight: .bold))
                                    .foregroundColor(.white)
                                
                                Text("Today")
                                    .font(.system(size: 12))
                                    .foregroundColor(.white.opacity(0.6))
                            }
                            
                            Spacer()
                            
                            ZStack {
                                Circle()
                                    .fill(Color.white.opacity(0.2))
                                    .frame(width: 56, height: 56)
                                
                                Image(systemName: "bell.fill")
                                    .resizable()
                                    .aspectRatio(contentMode: .fit)
                                    .frame(width: 24, height: 24)
                                    .foregroundColor(.white)
                            }
                        }
                        .padding(24)
                    }
                    .background(Color(red: 0.369, green: 0.376, blue: 0.808)) // #5E60CE
                    .cornerRadius(24)
                    .shadow(color: Color.black.opacity(0.1), radius: 8, x: 0, y: 4)
                    .padding(.horizontal, 24)
                    
                    Spacer().frame(height: 32)
                    
                    // Scenes Section
                    Text("Scenes")
                        .font(.system(size: 20, weight: .bold))
                        .foregroundColor(.black)
                        .padding(.horizontal, 24)
                    
                    Spacer().frame(height: 16)
                    
                    ScrollView(.horizontal, showsIndicators: false) {
                        HStack(spacing: 12) {
                            // Leading padding for scroll view content
                            Spacer().frame(width: 24).padding(.trailing, -12)
                            
                            SceneButton(text: "Home", icon: "house.fill", isSelected: true)
                            SceneButton(text: "Away", icon: "lock.fill", isSelected: false)
                            SceneButton(text: "Sleep", icon: "star.fill", isSelected: false)
                            
                            // Trailing padding for scroll view content
                            Spacer().frame(width: 24).padding(.leading, -12)
                        }
                    }
                    
                    Spacer().frame(height: 32)
                    
                    // Devices Section
                    Text("Devices")
                        .font(.system(size: 20, weight: .bold))
                        .foregroundColor(.black)
                        .padding(.horizontal, 24)
                    
                    Spacer().frame(height: 16)
                    
                    // Devices Grid
                    HStack(alignment: .top, spacing: 16) {
                        VStack(spacing: 16) {
                            DeviceCard(
                                name: "Smart Lamp",
                                icon: "star.fill",
                                isActive: true,
                                accentColor: Color(red: 1.0, green: 0.757, blue: 0.027) // #FFC107
                            )
                            DeviceCard(
                                name: "Thermostat",
                                icon: "line.3.horizontal",
                                isActive: true,
                                accentColor: Color(red: 1.0, green: 0.439, blue: 0.263) // #FF7043
                            )
                        }
                        
                        VStack(spacing: 16) {
                            DeviceCard(
                                name: "Main Lock",
                                icon: "lock.fill",
                                isActive: false,
                                accentColor: Color(red: 0.298, green: 0.686, blue: 0.314) // #4CAF50
                            )
                            DeviceCard(
                                name: "Wi-Fi Router",
                                icon: "gearshape.fill",
                                isActive: true,
                                accentColor: Color(red: 0.129, green: 0.588, blue: 0.953) // #2196F3
                            )
                        }
                    }
                    .padding(.horizontal, 24)
                    
                    Spacer().frame(height: 24)
                }
                .padding(.vertical, 24)
            }
        }
    }
}

struct SceneButton: View {
    let text: String
    let icon: String
    let isSelected: BooleanLiteralType
    
    var body: some View {
        HStack(spacing: 8) {
            Image(systemName: icon)
                .resizable()
                .aspectRatio(contentMode: .fit)
                .frame(width: 18, height: 18)
                .foregroundColor(isSelected ? .white : .black)
            
            Text(text)
                .font(.system(size: 14, weight: .medium))
                .foregroundColor(isSelected ? .white : .black)
        }
        .padding(.horizontal, 24)
        .padding(.vertical, 12)
        .background(isSelected ? Color(red: 0.129, green: 0.129, blue: 0.129) : Color.white) // #212121
        .cornerRadius(30)
    }
}

struct DeviceCard: View {
    let name: String
    let icon: String
    let isActive: Bool
    let accentColor: Color
    
    @State private var switchState: Bool
    
    init(name: String, icon: String, isActive: Bool, accentColor: Color) {
        self.name = name
        self.icon = icon
        self.isActive = isActive
        self.accentColor = accentColor
        _switchState = State(initialValue: isActive)
    }
    
    var body: some View {
        VStack(alignment: .leading) {
            HStack {
                ZStack {
                    Circle()
                        .fill(switchState ? accentColor.opacity(0.1) : Color(red: 0.941, green: 0.941, blue: 0.941)) // #F0F0F0
                        .frame(width: 40, height: 40)
                    
                    Image(systemName: icon)
                        .resizable()
                        .aspectRatio(contentMode: .fit)
                        .frame(width: 24, height: 24)
                        .foregroundColor(switchState ? accentColor : .gray)
                }
                Spacer()
            }
            
            Spacer()
            
            VStack(alignment: .leading, spacing: 4) {
                Text(name)
                    .font(.system(size: 16, weight: .bold))
                    .foregroundColor(.black)
                
                Text(switchState ? "On" : "Off")
                    .font(.system(size: 14, weight: .medium))
                    .foregroundColor(switchState ? accentColor : .gray)
            }
            
            Spacer()
            
            HStack {
                Spacer()
                Toggle("", isOn: $switchState)
                    .labelsHidden()
                    .toggleStyle(SwitchToggleStyle(tint: accentColor))
                    .scaleEffect(0.8)
            }
        }
        .padding(16)
        .frame(maxWidth: .infinity)
        // Aspect ratio 0.85 means width/height = 0.85, so height = width / 0.85
        // In a grid context, aspect ratio modifier works well.
        .aspectRatio(0.85, contentMode: .fit)
        .background(Color.white)
        .cornerRadius(20)
        .shadow(color: Color.black.opacity(0.05), radius: 2, x: 0, y: 1)
    }
}